packages/backend/src/connector/core/middleware/balance.ts

```diff
@@ -48,6 +49,22 @@ export function createBalanceMiddleware(): ILPMiddleware {
         case AccountTransferError.InsufficientBalance:
         case AccountTransferError.InsufficientLiquidity:
           throw new InsufficientLiquidityError(trxOrError)
+        case AccountTransferError.ReceiveLimitExceeded: {
+          const receivedAmount = destinationAmountOrError.toString()
+          assert.ok(accounts.outgoing.id)
+          const receiveLimit = await services.accounts.getReceiveLimit(
```

Comment:
The account's receive limit balance is a tigerbeetle debit account (`credits_must_not_exceed_debits`).
`calculateBalance` doesn't take `credits_reserved` into account (for either type), in which case I think the receive limit is kept further from `0`.

Reply:
Could you add an assertion that `receiveLimit !== 0` here just in case (maybe with your explanation as a comment)?

Reply:
:thinking: Could `receiveLimit` legitimately be `0` here though?
I have the accounts middleware checking for deactivated invoices, but could an invoice either not yet be deactivated by the worker, or could the invoice receive its remaining allowed amount between the accounts and balances middlewares?
https://github.com/interledger/rafiki/blob/bw-tigerbeetle-account/packages/backend/src/connector/core/middleware/account.ts#L23

Reply:
> Could receiveLimit legitimately be 0 here though?

You're right, I hadn't thought of that. Right now ilp-pay interprets an `F08` with `maximumAmount=0` as a connector error (correctly, I think) and terminates the connection. If the `receiveLimit` is legitimately `0`, we should probably use a different error code than `F08`, if for no other reason than to make it clearer to the sender that this isn't a connector bug. Maybe `F07`? Or since this particular bug is more likely being run into by a stream client sending prepares in parallel, maybe we should send a temporary error to avoid terminating their connection while the fulfill (presumably) races back. Full circle! :upside_down_face:

Reply:
I'll go with `F07`